pkg/upgrade/upgrades/v25_1_add_jobs_tables.go

```diff
@@ -95,57 +94,76 @@ func backfillJobsTablesAndColumns(
 	log.Infof(ctx, "backfilling new jobs tables and columns")
 	jobsBackfilled := 0
 	for {
+		candidateRows, err := d.DB.Executor().QueryBufferedEx(ctx, "jobs-backfill-find", nil, sessiondata.NodeUserSessionDataOverride,
+			`SELECT id FROM system.jobs WHERE owner IS NULL LIMIT $1`, jobsBackfillPageSize)
+		if err != nil {
+			return err
+		}
+		if len(candidateRows) == 0 {
+			// All done!
+			break
+		}
+
+		candidates := make([]int, len(candidateRows))
+		for i := range candidateRows {
+			candidates[i] = int(tree.MustBeDInt(candidateRows[i][0]))
+		}
 		var done int
 		if err := d.DB.DescsTxn(ctx, func(ctx context.Context, tx descs.Txn) (retErr error) {
-			// Find rows that have not been backfilled, which we will detect via a
-			// NULL owner. The coalesce() here isn't necessary since the vtable will
-			// produce empty strings rather than nulls if a job somehow is missing an
-			// owner, but since the owner not being null is what breaks the loop the
-			// extra coalesce here just ensures wouldn't loop forever if that were to
-			// change out from under us.
-			q, err := tx.QueryIteratorEx(ctx, "jobs-backfill-read", tx.KV(), sessiondata.NodeUserSessionDataOverride,
-				`SELECT
-				v.job_id,
-				v.description,
-				coalesce(v.user_name, ''),
-				v.finished,
-				v.error,
-				v.running_status,
-				v.fraction_completed,
-				v.high_water_timestamp
-				FROM crdb_internal.jobs v
-				LEFT JOIN system.jobs j ON j.id = v.job_id
-				WHERE j.owner IS NULL
-				LIMIT $1`, jobsBackfillPageSize,
-			)
+			if _, err := tx.ExecEx(ctx, "jobs-backfill-upgrade-locking", tx.KV(), sessiondata.NodeUserSessionDataOverride,
+				"SET enable_durable_locking_for_serializable = true"); err != nil {
```

Comment:
do we need this anymore?

Reply:
yes, if ranges move we don't want to abort.